src/com/amazon/ion/impl/bin/IonRawBinaryWriter.java

```diff
@@ -1686,6 +1616,8 @@ public void close() throws IOException
             buffer.close();
             patchBuffer.close();
             allocator.close();
+            // We cannot use `utf8StringEncoder` again after returning it to the pool.
+            Utf8StringEncoderPool.getInstance().returnEncoderToPool(utf8StringEncoder);
```

Comment:
When the writer is `close()`d, return the `Utf8StringEncoder` to the pool.

Reply:
I agree with both of the considerations raised here.